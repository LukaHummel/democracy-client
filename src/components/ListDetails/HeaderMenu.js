import React, { Component } from 'react';
import styled from 'styled-components/native';
import { Icon } from 'react-native-elements';

const Wrapper = styled.View`
  height: 35;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const ItemWrapper = styled.View`
  width: 50;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const MarkerIcon = styled(Icon).attrs({
  name: 'place',
})``;
const MarkerText = styled.Text``;

const CommentsIcon = styled(Icon).attrs({
  name: 'format-align-left',
})``;
const CommentText = styled.Text``;

const Date = styled.Text``;

export default class ListDetailsHeaderMenu extends Component {
  render() {
    return (
      <Wrapper>
        <ItemWrapper>
          <MarkerIcon />
          <MarkerText>234</MarkerText>
        </ItemWrapper>
        <ItemWrapper>
          <CommentsIcon />
          <CommentText>74</CommentText>
        </ItemWrapper>
        <ItemWrapper>
          <Date>3:21</Date>
        </ItemWrapper>
      </Wrapper>
    );
  }
}
