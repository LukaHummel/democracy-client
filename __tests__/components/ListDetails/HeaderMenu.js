import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import HeaderMenu from '../../../src/components/ListDetails/HeaderMenu';

describe('test components/List/Row', () => {
  it('render row', () => {
    const wrapper = renderer.create(<HeaderMenu commentsCount={35} places={75} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
