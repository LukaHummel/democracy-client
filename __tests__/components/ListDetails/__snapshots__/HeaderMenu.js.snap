// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 50,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontSize": 24,
          },
          Array [
            Object {},
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontSize": 12,
            "letterSpacing": -0.26,
          },
          undefined,
        ]
      }
    >
      75
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 50,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontSize": 24,
          },
          Array [
            Object {},
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontSize": 12,
            "letterSpacing": -0.26,
          },
          undefined,
        ]
      }
    >
      35
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 50,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#44db5e",
            "fontSize": 12,
            "letterSpacing": -0.26,
          },
          undefined,
        ]
      }
    >
      3:21
    </Text>
  </View>
</View>
`;
